packages/site-client/src/shared/cem/cem-class-declaration.ts

```diff
@@ -0,0 +1,292 @@
+  return html`
+    <tr>
+      <td>
+        <code
+          >${renderMemberName(
+            name,
+            deprecated,
+            'public',
+            false,
+            optional,
+            rest
+          )}</code
+        >
+      </td>
+      <td>
+        ${whenDefined(type, (t) => html`<cem-type .type=${t}></cem-type>`)}
+      </td>
+      <td>${markdown(summary)}${markdown(description)}</td>
+      <td>${whenDefined(def, (d) => html`<code>${d}</code>`)}</td>
+    </tr>
+  `;
+};
+
+const renderClassMethod = (method: cem.ClassMethod) => {
+  const {
+    name,
+    privacy,
+    static: statik,
+    deprecated,
+    description,
+    summary,
+    return: ret,
+    parameters,
+    // TODO(kschaaf) Not implemented in CEM yet
+    // source,
+    // inheritedFrom,
+  } = method;
+  return html`<tr>
+    <td>${renderMemberName(name, deprecated, privacy, statik)}</td>
+    <td>${markdown(summary)}${markdown(description)}</td>
+    <td>
+      ${whenDefined(
+        parameters,
+        (params: cem.Parameter[]) =>
+          html`
+            <table>
+              <thead>
+                <tr>
+                  <th>Name</th>
+                  <th>Type</th>
+                  <th>Description</th>
+                  <th>Default</th>
+                </tr>
+              </thead>
+              <tbody>
+                ${params.map(renderParameter)}
+              </tbody>
+            </table>
+          `
+      )}
+    </td>
+    <td>
+      ${whenDefined(
+        ret?.type,
+        (t) =>
+          html`Type: <cem-type .type=${t}></cem-type>${markdown(
+              ret?.summary
+            )}${markdown(ret?.description)}`
+      )}
+    </td>
+  </tr>`;
+};
+
+const renderSlot = (slot: cem.Slot) => {
+  return html`<tr>
+    <td>${renderMemberName(slot.name, slot.deprecated)}}</td>
+    <td>${markdown(slot.summary)}${markdown(slot.description)}</td>
+  </tr>`;
+};
+
+const renderCssProperty = (prop: cem.CssCustomProperty) => {
+  return html`<tr>
+    <td>${renderMemberName(prop.name, prop.deprecated)}}</td>
+    <td>${whenDefined(prop.syntax)}</td>
+    <td>${markdown(prop.summary)}${markdown(prop.description)}</td>
+    <td>${whenDefined(prop.default)}</td>
+  </tr>`;
+};
+
+const renderCssPart = (part: cem.CssPart) => {
+  return html`<tr>
+    <td>${renderMemberName(part.name, part.deprecated)}}</td>
+    <td>${markdown(part.summary)}${markdown(part.description)}</td>
+  </tr>`;
+};
+
+@customElement('cem-class-declaration')
+export class CemClassDeclaration extends LitElement {
+  static styles = styles;
+  @property()
+  declaration!: cem.ClassDeclaration;
+  @property()
+  exportName?: string;
+  render() {
+    return html`
+      ${renderDeclarationInfo(this.declaration, this.exportName)}
+      <slot name="usage"></slot>
+      ${whenDefined(
+        this.declaration.members?.filter(
+          (m) => m.kind === 'field' && m.privacy !== 'private'
+        ),
+        (m) => html` <h4>Fields</h4>
```

Comment:
nit: Prettier maintains leading whitespace if you initially had a newline. You can search for `` ` to find and remove these.